### PR TITLE
INTMDB-319: Add support for parameter unordered in resource_mongodbatlas_event_trigger

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

Add support for parameter unordered in resource_mongodbatlas_event_trigger

Link to any related issue(s):
https://jira.mongodb.org/browse/INTMDB-319
## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: website/docs/d/event_trigger.html.markdown [hunk 1]
```diff
@@ -64,6 +64,7 @@ In addition to all arguments above, the following attributes are exported:
 * `config_match` - A [$match](https://docs.mongodb.com/manual/reference/operator/aggregation/match/) expression document that MongoDB Realm includes in the underlying change stream pipeline for the trigger. 
 * `config_project` - A [$project](https://docs.mongodb.com/manual/reference/operator/aggregation/project/) expression document that Realm uses to filter the fields that appear in change event objects. 
 * `config_full_document` - If true, indicates that `UPDATE` change events should include the most current [majority-committed](https://docs.mongodb.com/manual/reference/read-concern-majority/) version of the modified document in the fullDocument field.
+* `unordered` - Sort order for `DATABASE` type.
```
Comment (themantissa) [inline website/docs/d/event_trigger.html.markdown]: Sorry I didn't get to you on this but the docs have a better explanation: from https://www.mongodb.com/docs/atlas/app-services/admin/api/v3/#operation/adminCreateTrigger   
```suggestion
* `unordered` - Only Available for Database Triggers. If true, event ordering is disabled and this trigger can process events in parallel. If false, event ordering is enabled and the trigger executes serially.
```

File: website/docs/d/event_triggers.html.markdown [hunk 2]
```diff
@@ -66,6 +66,7 @@ In addition to all arguments above, the following attributes are exported:
 * `config_match` - A [$match](https://docs.mongodb.com/manual/reference/operator/aggregation/match/) expression document that MongoDB Realm includes in the underlying change stream pipeline for the trigger.
 * `config_project` - A [$project](https://docs.mongodb.com/manual/reference/operator/aggregation/project/) expression document that Realm uses to filter the fields that appear in change event objects.
 * `config_full_document` - If true, indicates that `UPDATE` change events should include the most current [majority-committed](https://docs.mongodb.com/manual/reference/read-concern-majority/) version of the modified document in the fullDocument field.
+* `unordered` - Sort order for `DATABASE` type.
```
Comment (themantissa) [inline website/docs/d/event_triggers.html.markdown:69]:
```suggestion
* `unordered` - Only Available for Database Triggers. If true, event ordering is disabled and this trigger can process events in parallel. If false, event ordering is enabled and the trigger executes serially.
```

File: website/docs/r/event_trigger.html.markdown [hunk 3]
```diff
@@ -116,6 +117,7 @@ resource "mongodbatlas_event_trigger" "test" {
 * `config_match` - (Optional) Optional for `DATABASE` type. A [$match](https://docs.mongodb.com/manual/reference/operator/aggregation/match/) expression document that MongoDB Realm includes in the underlying change stream pipeline for the trigger. This is useful when you want to filter change events beyond their operation type. The trigger will only fire if the expression evaluates to true for a given change event.
 * `config_project` - (Optional) Optional for `DATABASE` type. A [$project](https://docs.mongodb.com/manual/reference/operator/aggregation/project/) expression document that Realm uses to filter the fields that appear in change event objects.
 * `config_full_document` - (Optional) Optional for `DATABASE` type. If true, indicates that `UPDATE` change events should include the most current [majority-committed](https://docs.mongodb.com/manual/reference/read-concern-majority/) version of the modified document in the fullDocument field.
+* `unordered` - (Optional) Optional for `DATABASE` type.
```
Comment (themantissa) [inline website/docs/r/event_trigger.html.markdown]:
```suggestion
* `unordered` - (Optional) Optional for for Database Triggers. If true, event ordering is disabled and this trigger can process events in parallel. If false, event ordering is enabled and the trigger executes serially.
```

**Review by themantissa — CHANGES_REQUESTED** [review]
Just a few docs changes, mostly since I didn't get you this in slack sooner :)

File: website/docs/r/event_trigger.html.markdown (hunk 3 above)
Comment (martinstibbe) [reply website/docs/r/event_trigger.html.markdown]: Updated

File: website/docs/d/event_triggers.html.markdown (hunk 2 above)
Comment (martinstibbe) [reply website/docs/d/event_triggers.html.markdown:69]: Updated

File: mongodbatlas/data_source_mongodbatlas_event_triggers.go [hunk 4]
```diff
@@ -190,6 +194,7 @@ func flattenEventTriggers(eventTriggers []realm.EventTrigger) []map[string]inter
 				"config_schedule":             eventTriggers[i].Config.Schedule,
 				"config_schedule_type":        eventTriggers[i].Config.ScheduleType,
 				"event_processors":            flattenTriggerEventProcessorAWSEventBridge(eventTriggers[i].EventProcessors),
+				"unordered":                   eventTriggers[i].Config.Unordered,
```
Comment (andreaangiolillo) [inline mongodbatlas/data_source_mongodbatlas_event_triggers.go:197]: [q] Could you update the tests to use `unordered`?

**Review by andreaangiolillo — APPROVED** [review]
Just one comment.

File: mongodbatlas/data_source_mongodbatlas_event_triggers.go (hunk 4 above)
Comment (martinstibbe) [reply mongodbatlas/data_source_mongodbatlas_event_triggers.go:197]: Added

File: website/docs/d/event_trigger.html.markdown (hunk 1 above)
Comment (martinstibbe) [reply website/docs/d/event_trigger.html.markdown]: Added new text description

**Review by andreaangiolillo — APPROVED** [review]

**Review by themantissa — APPROVED** [review]
Thank you for the fixes, LGTM!

**Pull request merged**